### PR TITLE
Clarify types and requirements levels for the plate/well specification

### PR DESCRIPTION
**Pull request by sbesson**
This commit rewrites the section and ensures that:

- each key has a requirement level specified as MUST/SHOULD/MAY
- the type for the value of each key is explicitly defined

The requirements & types should match the JSON schemas introduced in #120 

Staged at http://api.csswg.org/bikeshed/?url=https://raw.githubusercontent.com/sbesson/ngff/plate_specification_2/0.4/index.bs#plate-md

### CODE REVIEW

File: 0.4/index.bs [hunk 1]
```diff
@@ -437,59 +437,56 @@ above).
...
-      the `rows` list and a `columnIndex` key indentifying the index into
-      the `columns` list. `rowIndex` and `columnIndex` MUST be 0-based.
-      The `rowIndex`, `columnIndex`, and `path` MUST all refer to the same
-      row/column pair.</dd>
-</dl>
+custom attributes of the plate group under the `plate` key in the group-level metadata.
+
+The `plate` dictionary MAY contain an `acquisitions` key whose value MUST be a list of
+JSON objects defining the acquisitions for a given plate to which wells can refer to. Each
+acquisition object MUST contain an `id` key whose value MUST be an unique integer identifier
```
Comment (melissalinkert) [inline 0.4/index.bs:444]: `a unique integer identifier greater than or equal to 0`?

File: 0.4/index.bs [hunk 2]
```diff
@@ -437,59 +437,56 @@ above).
...
+within the context of the plate to which fields of view can refer to (see #well-md).
+Each acquisition object SHOULD contain a `name` key whose value MUST be a string identifying
+the name of the acquisition. Each acquisition object SHOULD contain a `maximumfieldcount`
+key whose value MUST be an integer indicating the maximum number of fields of view for the
+acquisition. Each acquisition object MAY contain a `description` key whose value MUST be a
```
Comment (melissalinkert) [inline 0.4/index.bs:449]: Is it expected that `description` isn't in the plate JSON schema?

File: 0.4/index.bs [hunk 3]
```diff
@@ -437,59 +437,56 @@ above).
...
-</dl>
+custom attributes of the plate group under the `plate` key in the group-level metadata.
+
+The `plate` dictionary MAY contain an `acquisitions` key whose value MUST be a list of
+JSON objects defining the acquisitions for a given plate to which wells can refer to. Each
+acquisition object MUST contain an `id` key whose value MUST be an unique integer identifier
+within the context of the plate to which fields of view can refer to (see #well-md).
+Each acquisition object SHOULD contain a `name` key whose value MUST be a string identifying
+the name of the acquisition. Each acquisition object SHOULD contain a `maximumfieldcount`
+key whose value MUST be an integer indicating the maximum number of fields of view for the
```
Comment (melissalinkert) [inline 0.4/index.bs:448]: `an integer greater than 0`?

File: 0.4/index.bs [hunk 4]
```diff
@@ -437,59 +437,56 @@ above).
...
+acquisition. Each acquisition object MAY contain a `description` key whose value MUST be a
+string specifying a description for the acquisition. Each acquisition object MAY contain
+a `startime` and/or `endtime` key whose values MUST be integer epoch timestamps specifying
```
Comment (melissalinkert) [inline 0.4/index.bs:451]: `starttime`

File: 0.4/index.bs [hunk 5]
```diff
@@ -437,59 +437,56 @@ above).
...
+the start and/or end timestamp of the acquisition.
+
+The `plate` dictionary MUST contain a `columns` key whose value MUST be a list of JSON objects
+defining the columns of the plate. Each column object object defines the properties of
```
Comment (melissalinkert) [inline 0.4/index.bs:455]: `column object defines`

File: 0.4/index.bs [hunk 6]
```diff
@@ -437,59 +437,56 @@ above).
...
+the column at the index of the object in the list. Each column in the physical plate
+MUST be defined, even if no wells in the column are defined. Each column object MUST
+contain a `name` key whole value is a string specifying the column name. The `name` MUST
```
Comment (melissalinkert) [inline 0.4/index.bs:458]: `whose value`

File: 0.4/index.bs [hunk 7]
```diff
@@ -437,59 +437,56 @@ above).
...
+contain only alphanumeric characters, MUST be case-sensitive, and MUST NOT be a duplicate of any
+other `name` in the `columns` list. Care SHOULD be taken to avoid collisions on
+case-insensitive filesystems (e.g. avoid using both `Aa` and `aA`).
+
+The `plate` dictionary SHOULD contain a `field_count` key whose value MUST be an integer
```
Comment (melissalinkert) [inline 0.4/index.bs:463]: `an integer greater than 0`?

File: 0.4/index.bs [hunk 8]
```diff
@@ -437,59 +437,56 @@ above).
...
+defining the maximum number of fields per view across all wells.
+
+The `plate` dictionary SHOULD contain a `name` key whose value MUST be a string defining the
+name of the plate.
+
+The `plate` dictionary MUST contain a `rows` key whose value MUST be a list of JSON objects
+defining the rows of the plate. Each row object object defines the properties of
```
Comment (melissalinkert) [inline 0.4/index.bs:470]: `row object defines`

File: 0.4/index.bs [hunk 9]
```diff
@@ -514,17 +511,15 @@ For high-content screening datasets, the metadata about all fields of views
...
-      field of view. If multiple acquisitions were performed in the plate, it
-      MUST contain an `acquisition` key identifying the id of the
-      acquisition which must match one of acquisition JSON objects defined in
-      the plate metadata.</dd>
-  <dt><strong>version</strong></dt>
-  <dd>A string defining the version of the specification.</dd>
-</dl>
+The `well` dictionary MUST contain an `images` key whose value MUST be a list of JSON objects
+specifying all fields of views for a given well. Each image object MUST contain a
+`path` key whose value MUST be a string specifying the path to the field of view. If multiple
```
Comment (melissalinkert) [inline 0.4/index.bs:516]: Add something like ```The `path` MUST
contain only alphanumeric characters, MUST be case-sensitive, and MUST NOT be a duplicate of any
other `path` in the `images` list.```, similar to row and column names above?

File: 0.4/index.bs [hunk 10]
```diff
@@ -514,17 +511,15 @@ For high-content screening datasets, the metadata about all fields of views
...
+acquisitions were performed in the plate, it MUST contain an `acquisition` key whose value
+MUST be an integer identifying the acquisition which MUST match one of acquisition JSON
```
Comment (melissalinkert) [inline 0.4/index.bs:518]: `one of the`

**Review by melissalinkert — CHANGES_REQUESTED** [review]
Super pedantic comments aside (some of which are things not introduced by this PR), this does read better than before.

**Comment by sbesson** [comment]
Thanks for the thorough review @melissalinkert. Pedantic comments are very useful. The aim of this PR is to reduce the ambiguity in the definition of these specifications. Pushed a few commits addressing your suggestions and adding the acquisition description to the JSON schema.

I also ported the same text to the `latest` specification to save myself another PR once this is merged. If additional changes are requested, I'll apply them to both specifications.

File: 0.4/index.bs [hunk 11]
```diff
@@ -442,32 +442,33 @@ custom attributes of the plate group under the `plate` key in the group-level me
 The `plate` dictionary MAY contain an `acquisitions` key whose value MUST be a list of
 JSON objects defining the acquisitions for a given plate to which wells can refer to. Each
 acquisition object MUST contain an `id` key whose value MUST be an unique integer identifier
-within the context of the plate to which fields of view can refer to (see #well-md).
+greater than of equal to 0 within the context of the plate to which fields of view can refer
```
Comment (melissalinkert) [inline 0.4/index.bs:445]: `or equal to`

File: latest/index.bs [hunk 12]
```diff
@@ -439,59 +439,57 @@ above).
...
-      the `columns` list. `rowIndex` and `columnIndex` MUST be 0-based.
-      The `rowIndex`, `columnIndex`, and `path` MUST all refer to the same
-      row/column pair.</dd>
-</dl>
+custom attributes of the plate group under the `plate` key in the group-level metadata.
+
+The `plate` dictionary MAY contain an `acquisitions` key whose value MUST be a list of
+JSON objects defining the acquisitions for a given plate to which wells can refer to. Each
+acquisition object MUST contain an `id` key whose value MUST be an unique integer identifier
+greater than of equal to 0 within the context of the plate to which fields of view can refer
```
Comment (melissalinkert) [inline latest/index.bs:447]: `or equal to`

File: latest/index.bs [hunk 13]
```diff
@@ -439,59 +439,57 @@ above).
...
+defining the rows of the plate. Each row object defines the properties of
+the row at the index of the object in the list. Each row in the physical plate
+MUST be defined, even if no wells in the row are defined. Each defined row MUST
+contain a `name` key whose value MUST be a string defining the row name. The `name` MUST
+contain only alphanumeric characters, MUST be case-sensitive, and MUST NOT be a duplicate of any
+other `name` in the `rows` list. Care SHOULD be taken to avoid collisions on
+case-insensitive filesystems (e.g. avoid using both `Aa` and `aA`).
+
+The `plate` dictionary SHOULD contain a `version` key whose value MUST be a string specifying the
+version of the plate specificaton.
```
Comment (melissalinkert) [inline latest/index.bs:482]: `specification` (as fixed above)

**Review by melissalinkert — APPROVED** [review]
Just a couple of typos left, :+1: once those are fixed.

**Pull request merged**